nix/gecko_env.nix

```diff
@@ -78,6 +81,16 @@ in gecko.overrideDerivation (old: {
     ac_add_options --with-clang-path=${clang_4}/bin/clang
     ac_add_options --with-libclang-path=${llvmPackages_4.libclang}/lib
     mk_add_options AUTOCLOBBER=1
+
+    # Build Firefox for Android:
+    ac_add_options --enable-application=mobile/android
+    ac_add_options --target=arm-linux-androideabi
```

Comment:
Does this options looks OK to you @marco-c  & @abpostelnicu  ?
This will not *only* build the android version, right ?

Reply:
yes, this is ok

Reply:
Thanks boss.

Reply:
Why do we specify the mobile application if we don't know yet that we have java files? This doesn't seem right, we are configuring an Android build and when we will generate the compile_commands.json we will have declarations related for Android so hence we will run clang-tidy for an Android build.
It would be nice to avoid this.

Reply:
I have a workaround for this, but it will involve running `configure` and `export` two times.

Reply:
I was thinking of having two separate mozconfigs. One that is generated by nix, and another one which I generate from within python.